### ISSUE TITLE
[Research] Channel Reflection Data Augmentation for Few-Channel EEG Training (+3-8% accuracy)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#37]
## Paper
**Title:** Channel Reflection: Knowledge-Driven Data Augmentation for EEG-Based Brain-Computer Interfaces
**Authors:** Ziwei Wang, Siyang Li, Jingwei Luo, Jiajing Liu, Dongrui Wu
**Year:** 2024
**Venue:** Neural Networks (Elsevier) / arXiv 2412.03224
**URL:** https://arxiv.org/abs/2412.03224

## Key Contribution
A **parameter-free** data augmentation method that exploits the hemispheric symmetry of EEG channel layouts. For each training sample, a mirrored version is created by swapping left/right symmetric channels (e.g., AF7 <-> AF8, TP9 <-> TP10). Tested on **8 public datasets across 4 BCI paradigms** (motor imagery, SSVEP, P300, seizure). Consistently outperforms existing augmentation methods (noise injection, time warping, GAN-based). Key: it uses neurophysiological prior knowledge — emotional states have lateralized neural correlates, so swapping hemispheres creates valid augmented samples.

## Relevance to NeuralDreamWorkshop
Muse 2's 4 channels are perfectly symmetric: TP9/AF7 (left) mirror TP10/AF8 (right). Channel reflection directly applies:
- AF7 <-> AF8 (frontal pair)
- TP9 <-> TP10 (temporal pair)

This **doubles the training dataset for free** with neurophysiologically valid augmented samples. For emotion classification, swapping hemispheres effectively inverts the FAA sign — a sample with positive FAA (happy) becomes one with negative FAA (sad-like), providing natural class balancing.

### Files to Create
- `ml/training/augmentation.py` — Channel reflection + complementary augmentations module:
  ```python
  def channel_reflect(eeg_4ch: np.ndarray) -> np.ndarray:
      """Swap AF7<->AF8 and TP9<->TP10."""
      reflected = eeg_4ch.copy()
      reflected[0], reflected[3] = eeg_4ch[3].copy(), eeg_4ch[0].copy()  # TP9<->TP10
      reflected[1], reflected[2] = eeg_4ch[2].copy(), eeg_4ch[1].copy()  # AF7<->AF8
      return reflected

  def augment_batch(batch, labels, methods=['reflect', 'noise', 'time_shift']):
      """Apply channel reflection + optional noise/shift augmentation."""
  ```

### Files to Modify
- `ml/training/mega_trainer.py` — Integrate channel reflection into training pipeline:
  - Before training, double dataset via `channel_reflect()` on every sample
  - For emotion labels: flip valence sign on reflected samples (FAA inverts)
  - Keep arousal label unchanged (bilateral signal)
- `ml/training/train_emotion.py` — Add augmentation flag `--augment-cr` (channel reflection)
- `ml/training/train_dreamer.py` — Apply CR augmentation to DREAMER training
- `ml/processing/eeg_processor.py` — Add `amplitude_noise(signal, std=0.05)` and `temporal_jitter(signal, max_shift=5)` as complementary augmentations for dry-electrode nuisances

### Augmentation Combinations (from paper)
```
1. Channel Reflection alone: +3-5% accuracy
2. CR + Gaussian noise (simulate impedance drift): +5-7%
3. CR + temporal shift (simulate movement jitter): +4-6%
4. CR + noise + shift (full stack): +5-8%
```

### Label Handling for Emotion
```
Original sample: AF7=left, AF8=right → FAA = ln(AF8_alpha) - ln(AF7_alpha) > 0 → happy
Reflected sample: AF7↔AF8 → FAA sign flips → label adjusted for valence flip
Arousal: unchanged (bilateral beta increase is symmetric)
Discrete emotions: happy↔sad, angry stays angry (approach motivation is bilateral)
```

## Expected Impact
- **Training data:** 2x increase (free, no new data collection)
- **Emotion accuracy:** +3-8% cross-subject (from 74.21% → 77-82%)
- **Sleep/dream accuracy:** +2-4% (less lateralized, but still benefits)
- **Key win:** Zero computational cost at inference — only used during training

## Complexity: LOW
- Core implementation is 10 lines of channel swapping
- No new dependencies
- No changes to inference pipeline
- Can be applied to all existing training scripts with a flag
